app/lib/account/session_cookie.dart

```diff
@@ -0,0 +1,91 @@
+/// Logic to do with session cookie parsing and reading.
+library session_cookie;
+
+import 'dart:io' show HttpDate, HttpHeaders;
+
+import '../shared/configuration.dart' show envConfig;
+import 'models.dart';
+
+/// The name of the session cookie.
+///
+/// Depends on whether we're running locally.
+String get _pubSessionCookieName {
+  if (envConfig.isRunningLocally) {
+    return 'pub-sid-insecure'; // Note. this should only happen on localhost.
+  }
+
+  // Cookies prefixed '__Host-' must:
+  //  * be set by a HTTPS response,
+  //  * not feature a 'Domain' directive, and,
+  //  * have 'Path=/' directive.
+  // Hence, such a cookie cannot have been set by another website or an
+  // HTTP proxy for this website.
+  return '__Host-pub-sid';
+}
+
+/// Create a set of HTTP headers that store a session cookie.
+Map<String, String> createSessionCookie(UserSessionData session) {
+  // Always create a cookie that expires 25 minutes before the session.
+  // This way clock skew on the client is less likely to cause us to receive
+  // an invalid cookie. Not that getting an expired cookie should be a problem.
+  final expiration = session.expires.subtract(Duration(minutes: 25));
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  return {
+    HttpHeaders.setCookieHeader: [
+      '$_pubSessionCookieName=${session.sessionId}',
+      // Send cookie to anything under '/' required by '__Host-' prefix.
+      'Path=/',
+      // Max-Age takes precedence over 'Expires', this also has the benefit of
+      // not being corrupted by client-side clock skew.
+      'Max-Age=${expiration.difference(DateTime.now()).inSeconds}',
+      // Cookie expires when the session expires.
+      'Expires=${HttpDate.format(expiration)}',
+      // Do not include the cookie in CORS requests, unless the request is a
+      // top-level navigation to the site, as recommended in:
+      // https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-02#section-8.8.2
+      'SameSite=Lax',
+      if (!envConfig.isRunningLocally)
+        'Secure', // Only allow this cookie to be sent when making HTTPS requests.
+      'HttpOnly', // Do not allow Javascript access to this cookie.
+    ].join('; '),
+  };
+}
+
+/// Parse [cookieString] and return `sessionId` or `null`.
+///
+/// The [cookieString] is the value of the `cookie:` request header.
+String parseSessionCookie(String cookieString) {
+  ArgumentError.checkNotNull(cookieString, 'cookieString');
+
+  // The cookieString is separated by '; ', and contains 'name=value'
+  // See: https://tools.ietf.org/html/rfc6265#section-5.4
+  final cookie = cookieString.split('; ').firstWhere(
```

Comment:
Nit: We have `parseCookieHeader` in `shared/utils.dart`.

Reply:
Okay, did a few tweaks and adopted this..